lib/src/chart/base/base_chart/base_chart_data.dart

```diff
@@ -78,6 +78,79 @@ class FlBorderData with EquatableMixin {
       ];
 }
 
+/// Holds data to drawing shdow for the chart.
+class FlShadowData with EquatableMixin {
+  /// [show] Determines showing or hiding border around the chart.
+  /// [offset] Determines the offset that shadow differ from the object.
+  /// [color] Determines the shadow color.
+  /// [maskFilter] Determines the mask filter will be applied to visual the shadow.
+  /// [numberOfLayers] Determines number of layers that will be painted (it should be 1 layer in case of normal shadow).
+  /// [minDarkenValue] Determines the start value of darken will be applied on the color in case of multiple layers.
+  /// [maxDarkenValue] Determines the end value of darken will be applied on the color in case of multiple layers.
+  FlShadowData({
+    bool? show,
+    Offset? offset,
+    Color? color,
+    this.maskFilter,
+    int? numberOfLayers,
+    int? minDarkenValue,
+    int? maxDarkenValue,
+  })  : show = show ?? true,
+        offset = offset ?? const Offset(3, 3),
+        color = color ?? Colors.black.withOpacity(0.5),
+        numberOfLayers = numberOfLayers ?? 1,
+        minDarkenValue = minDarkenValue ?? 0,
```

Comment:
Please define a better value for minDarkenValue and maxDarkenValue to simplify the usage. (for example 40 and 70 as you used in examples)